src/core/models.py

```diff
@@ -419,14 +422,18 @@ def snapshot_self(self, article):
 
         if frozen_author:
             for k, v in frozen_dict.items():
+                frozen_dict['order'] = frozen_author.order
```

Comment:
This should be done outside of the loop, right after `if frozen_author:` above(unless I'm missing something)
It could also use a comment (if my understanding is correct, we want to prioritize `frozen_author.order` over `ArticleAuthorOrder` because the reordering tool will, after submission is completed, save the ordering to `frozen_author.order` only and not `ArticleAuthorOrder`)